openapi.yaml

```diff
@@ -626,6 +913,140 @@ components:
       additionalProperties: false
       required:
         - permissions
+    Constraints:
+      type: object
+      properties:
+        allocate-public-ip:
+          type: boolean
+        arch:
+          type: string
+        container:
+          type: string
+        cores:
+          type: integer
+        cpu-power:
+          type: integer
+        image-id:
+          type: string
+        instance-role:
+          type: string
+        instance-type:
+          type: string
+        mem:
+          type: integer
+        root-disk:
+          type: integer
+        root-disk-source:
+          type: string
+        spaces:
+          type: array
+          items:
+            type: string
+        tags:
+          type: array
+          items:
+            type: string
+        virt-type:
+          type: string
+        zones:
+          type: array
+          items:
+            type: string
+      additionalProperties: false
+    DeployFromRepositoryArgs:
+      type: object
+      properties:
+        AttachStorage:
+          type: array
+          items:
+            type: string
+        CharmName:
+          type: string
+        ConfigYAML:
+          type: string
+        Cons:
+          $ref: '#/components/schemas/Constraints'
+        Devices:
+          type: object
+          patternProperties:
+            .*:
+              $ref: '#/components/schemas/Constraints'
+        Placement:
+          type: array
+          items:
+            $ref: '#/components/schemas/Placement'
+        Storage:
+          type: object
+          patternProperties:
+            .*:
+              $ref: '#/components/schemas/Constraints'
+        base:
+          $ref: '#/components/schemas/Base'
+        channel:
+          type: string
+        endpoint-bindings:
+          type: object
+          patternProperties:
+            .*:
+              type: string
+        resources:
+          type: object
+          patternProperties:
+            .*:
+              type: string
+        revision:
+          type: integer
+      additionalProperties: false
+      required:
+        - CharmName
+    DeployFromRepositoryInfo:
+      type: object
+      properties:
+        architecture:
```

Comment:
I notice that in some places you call this `arch` and others `architecture`. Let's standardise on `architecture`?